Failing tests for go-1.11 on i686-linux

## Issue description

The build of go-1.11 on i686-linux is failing because one of the tests does not pass. We can see that happening on Hydra, for example:

https://hydra.nixos.org/build/80999166/nixlog/12

This issue has already been reported to go and is happening on some Debian 386 builds as well. For reference: https://github.com/golang/go/issues/25628

Note that there are multiple tests failing due to this error. So this issue is related to #45960.

### Steps to reproduce

```
nix-build -A go . --argstr system i686-linux
```

Results in:

```
--- FAIL: TestExtraFiles (0.08s)
    exec_test.go:607: Run: exit status 1; stdout "leaked parent file. fd = 44; want 43\n", stderr ""
FAIL
FAIL    os/exec 1.636s
```

Note that x86_64 does pass the tests.

A possible workaround for the moment is disabling checks for the go package for i686.

--- a/pkgs/development/compilers/go/1.11.nix
+++ b/pkgs/development/compilers/go/1.11.nix
@@ -125,6 +125,7 @@ stdenv.mkDerivation rec {
     ./remove-fhs-test-references.patch
     ./skip-external-network-tests.patch
     ./skip-nohup-tests.patch
+    ./skip-test-extra-files-on-386.patch
   ];
 
   postPatch = optionalString stdenv.isDarwin ''

--- a/pkgs/development/compilers/go/skip-test-extra-files-on-386.patch
+++ b/pkgs/development/compilers/go/skip-test-extra-files-on-386.patch
@@ -1,0 +1,15 @@
+diff --git a/src/os/exec/exec_test.go b/src/os/exec/exec_test.go
+index 558345ff63..22129bf022 100644
+--- a/src/os/exec/exec_test.go
++++ b/src/os/exec/exec_test.go
+@@ -593,6 +593,10 @@ func TestExtraFiles(t *testing.T) {
+ 		t.Skipf("skipping test on %q", runtime.GOOS)
+ 	}
+ 
++	if runtime.GOOS == "linux" && runtime.GOARCH  == "386" {
++		t.Skipf("skipping test on %q %q", runtime.GOARCH, runtime.GOOS)
++	}
++
+ 	// Ensure that file descriptors have not already been leaked into
+ 	// our environment.
+ 	if !testedAlreadyLeaked {

